lib/recog/fingerprint.rb

```diff
@@ -43,7 +47,17 @@ def initialize(xml)
   # @return [Hash,nil] Keys will be host, service, and os attributes
   def match(match_string)
     # match_string.force_encoding('BINARY') if match_string
-    match_data = @regex.match(match_string)
+    begin
+      match_data = @regex.match(match_string)
+    rescue Encoding::CompatibilityError => e
+      # 'ASCII-8BIT' encoded responses will be caught and fixed here.
+      match_data = @regex.match(match_string.force_encoding('UTF-8'))
```

Comment:
Handled, but my debugging code is still in place for the moment.  Pretty much everything I tested with today was ASCII-8BIT but the only time this was a problem was when (I guess?) the regex was trying to automatically change it to UTF-8.   I would just encode everything as UTF-8 but I'm not sure about the performance impacts yet.